mungegithub/mungers/submit-queue_test.go

```diff
@@ -902,6 +904,8 @@ func TestSubmitQueue(t *testing.T) {
 				// so we don't want to modify the PR there. Instead we need
 				// to wait until the second time we check Jenkins, which happens
 				// we did the IsMerged() check.
+				testChecksLock.Lock()
+				defer testChecksLock.Unlock()
```

Comment:
This is inside a handler function definition. The lock is not locked or unlocked inside the loop, instead handlers are registered inside the loop. The lock is used when the handler is called and the defer is activated when the handler returns.